servers/jax-rs/src/main/java/org/projectnessie/services/authz/AccessCheckerExtension.java

```diff
@@ -27,51 +28,99 @@
 
 /** This class needs to be in the same package as {@link AccessChecker}. */
 public class AccessCheckerExtension implements Extension {
-  public static final AccessChecker ACCESS_CHECKER =
+  private static final AccessChecker NOOP_ACCESS_CHECKER = new DefaultAccessChecker();
+
+  private volatile Supplier<AccessChecker> accessCheckerSupplier;
```

Comment:
Could this be `final`? It looks like test methods do not have access to this field directly and have to go through `NessieJaxRsExtension` :thinking:

Reply:
Ah, no.
It doesn't allow `AccessCheckerExtension` (or anything else) to have constructor arguments.